### ISSUE TITLE
Set up JavaScript testing framework and write initial tests

### ISSUE THREAD

**Issue by tobscure** [flarum/core#4]
Now that we're using Mithril, we have complete freedom over what we use and how we use it.

A good starting point might be to copy whatever Mithril does itself.

**Comment by tobscure** [comment]
OK I made a start on this too (at least I fixed JSHint errors) but I'm struggling to even get the first test running. All `index-test.js` is doing is calling `visit('/')` but it is still failing with the following unhelpful message:

```
    ---
        actual: >
            null
        message: >
            Error: Something went wrong
        Log: >
    ...
```

Can anyone lend a hand?

**Comment by tobyzerner** [comment]
Sorted it out — it was because the API URL was wrong. Fixed in d13ef6f1fc5d5cd79c44e37ab7853410049bccc0. Wrote some initial tests in bbf2c971e403a0e61f0718241b577fff6b9475d5. One is failing, waiting on kurko/ember-json-api#59

Ultimately we'll need to set up fixtures/synthetic AJAX responses rather than the calling a live API.

**Comment by tobyzerner** [comment]
Closing in favour of flarum/issue-archive#424
